lib/util/hud_code_gen.rb

```diff
@@ -0,0 +1,150 @@
+    skipped = []
+    filename = "lib/util/concerns/hud_lists_#{year}.rb"
+    arr = []
+    arr.push ::Code.copywright_header
+    arr.push "
+      # frozen_string_literal: true
+      #{CODEGEN_FILE_HEADER}
+      module Concerns::HudLists#{year}
+        extend ActiveSupport::Concern
+          class_methods do
+            def _translate(map, id, reverse)
+              if reverse
+                rx = forgiving_regex id
+                if rx.is_a?(Regexp)
+                  map.detect { |_, v| v.match?(rx) }.try(&:first)
+                else
+                  map.detect { |_, v| v == rx }.try(&:first)
+                end
+              else
+                map[id] || id
+              end
+            end
+    "
+
+    all_lists.each do |element|
+      next if skipped.include?(element['code'].to_s)
+
+      map_values = element['values'].map do |obj|
+        description = obj['description'].strip
+        "#{obj['key'].to_json} => \"#{description}\""
+      end.join(",\n")
+
+      map_name, lookup_fn_name = get_function_names(element['name'])
+
+      arr.push "
+      # #{element['code']}
+      def #{map_name}
+        {\n#{map_values}\n}.freeze
+      end
+
+      def #{lookup_fn_name}(id, reverse = false)
+        _translate #{map_name}, id, reverse
+      end
+      "
+    end
+    arr.push 'end'
+    arr.push 'end'
+    contents = arr.join("\n")
+    File.open(filename, 'w') do |f|
+      f.write(contents)
+    end
+    filename
+  end
+
+  def generate_graphql_enums(year = '2024')
+    source = File.read("lib/data/#{year}_hud_lists.json")
+    skipped = ['race', '3.6.1', '2.4.2', '1.6']
+    filename = 'drivers/hmis/app/graphql/types/hmis_schema/enums/hud.rb'
+    hud_utility_class = year == '2022' ? 'HudUtility' : "HudUtility#{year}"
+
+    seen = []
+    arr = []
+    arr.push ::Code.copywright_header
+    arr.push "
+      # frozen_string_literal: true
+      #{CODEGEN_FILE_HEADER}
+      module Types::HmisSchema::Enums::Hud
+    "
+    JSON.parse(source).each do |element|
+      next if skipped.include?(element['code'].to_s)
+
+      name = element['name']
+      next if seen.include?(name)
+
+      map_name = get_function_names(name)[0]
+
+      arr.push "  class #{name} < Types::BaseEnum"
+      arr.push "    description '#{element['code'] || name}'"
+      arr.push "    graphql_name '#{name}'"
+      arr.push "    hud_enum #{hud_utility_class}.#{map_name}"
+      arr.push '  end'
+      seen << name
+    end
+
+    arr.push 'end'
+    contents = arr.join("\n")
+    File.open(filename, 'w') do |f|
+      f.write(contents)
+    end
+    filename
+  end
+
+  private def get_function_names(name)
+    lookup_fn_name = name.underscore
+    map_name = lookup_fn_name.pluralize
+
+    # apply overrides
+    map_name = MAP_NAME_OVERRIDES[map_name] if MAP_NAME_OVERRIDES.key?(map_name)
+    lookup_fn_name = LOOKUP_FN_OVERRIDES[lookup_fn_name] if LOOKUP_FN_OVERRIDES.key?(lookup_fn_name)
```

Comment:
Do you think it makes sense to version these overrides by year as well?  It would mean we could make 2024 much more standard, but swapping methods out throughout the codebase would be harder.

Reply:
I don't think so.. the whole benefit of using overrides is that we can minimize our impact on the codebase when we do the switchover. `HudUtility=>HudUtility2024` is a much simpler replacement. We can take our time removing the overrides after that, or standardizing them to suffix with _options, or whatever.

Reply:
Yep, sounds good!